feat(vue-app): build indicator

pi0 opened the pull request:
<!--- Provide a general summary of your changes in the title above -->

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)


## Description

Show build progress inside pages!

![nuxt-hmr-indicator](https://user-images.githubusercontent.com/904724/58546148-10cf9e00-8205-11e9-995d-4453e3ed950d.gif)


## TODO
  - [x] improve style
  - [x] support custom base URL
  - [x] add an option to optionally disable
  - [x] wait little more to display progress of 100

## Checklist:
<!--- Put an `x` in all the boxes that apply. -->
<!--- If your change requires a documentation PR, please link it appropriately -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly. (PR: https://github.com/nuxt/docs/pull/1421)
- [ ] I have added tests to cover my changes (if not applicable, please state why)
- [x] All new and existing tests are passing.

Commit:
feat: inline HMR progress indicator

Commit:
support router base

Commit:
fix nuxt err

Atinux commented:
Awesome work Pooya!
Will do some small changes for the UI and will be good for me :)

clarkdo commented on packages/vue-app/template/components/nuxt-hmr.vue:
```diff
@@ -0,0 +1,103 @@
+<template>
+<transition appear>
+  <div class="nuxt__hmr" v-if="building">
+      <div>Building ({{ progress}}%)</div>
```
`space`

pi0 replied:
fixed

clarkdo commented on packages/vue-app/template/components/nuxt-hmr.vue:
```diff
@@ -0,0 +1,103 @@
+
+    wsClose() {
+        if (this.ws) {
+        this.ws.close()
```
indentation

clarkdo commented on packages/vue-app/template/components/nuxt-hmr.vue:
```diff
@@ -0,0 +1,103 @@
+        if (data[0] === '{') {
+          data = JSON.parse(data)
+        }
+      } catch (e) {}
```
If data is not json, is following code supposed to still continue executing?

The `allDone` and `states` are all `undefined` which will cause exception as well.

clarkdo commented on packages/vue-app/template/components/nuxt.js:
```diff
@@ -12,12 +12,15 @@ import NuxtError from './nuxt-error.vue'
 <% } %>
 import NuxtChild from './nuxt-child'
 
+<%if (isDev) { %>import NuxtHMR from './nuxt-hmr'<% } %>
```
Also wrapped with `client` ?

pi0 replied:
This component is SSR safe and renders an empty node (only for dev)

clarkdo commented on packages/vue-app/template/components/nuxt-hmr.vue:
```diff
@@ -0,0 +1,103 @@
+
+    wsReconnect(e) {
+      setTimeout(() => {
+        this.wsConnect()
```
Will this lead to infinite re-connect ?

pi0 replied:
i'll add a limit of 10

Commit:
fix space

Commit:
fix indentation

Commit:
return in case of ws message parsing error

Commit:
close ws on beforeDestroy

Commit:
ui: Update loading indicator UI

Atinux commented:
My last commit improves the style and uses `$nextTick` to show the 100% everytime.

![nuxt-hmr-indicator](https://user-images.githubusercontent.com/904724/58457849-fbd10d00-8127-11e9-8b98-94d0630a3004.gif)

Atinux commented:
About the option to disable it, what about `build.indicator: false`?

Commit:
builder: Add build.indicator option

kevinmarrec commented on packages/config/types/build.d.ts:
```diff
@@ -28,6 +28,7 @@ interface NuxtBabelOptions extends Pick<TransformOptions, Exclude<keyof Transfor
 
 export interface NuxtConfigurationBuild {
   analyze?: BundleAnalyzerPlugin.Options | boolean
+  indicator?: boolean
```
Move `indicator` below (between `html` & `loaders`) so it fits alphabetical consistency

Atinux replied:
Fixed, thanks Kevin, I don't know how I missed this!

kevinmarrec reviewed (dismissed):
Awesome work guys :) !

@Atinux I'm fine with `build.indicator`, unless we decide it needs more context.

Commit:
ui: Use only logo and %

Commit:
hotfix: Alphabetical order

Atinux commented:
UI changes to show only Nuxt logo + percent:
![nuxt-hmr-indicator](https://user-images.githubusercontent.com/904724/58462388-1314f800-8132-11e9-818a-43bf8b45721b.gif)

Commit:
hotfix: Add fixed with and add back v-if

Commit:
minor style change

Commit:
rename component to build-indicator

Commit:
feat: animated progress

Commit:
assign name to component

Commit:
update test

pi0 commented:
Updates:
- Added name to component and renamed to `buildIndicator`
- Smooth progress animation between updates

![anim](https://user-images.githubusercontent.com/5158436/58500509-93ba0f80-8197-11e9-8524-e115c6d32571.gif)

Commit:
naming consistency

Commit:
render into app to prevent dom wrapping

Commit:
extra new line

Commit:
better App.js formatting

Commit:
update snapshot

Commit:
clear interval

Atinux commented:
Final GIF

![nuxt-hmr-indicator](https://user-images.githubusercontent.com/904724/58546148-10cf9e00-8205-11e9-995d-4453e3ed950d.gif)

Commit:
Merge branch 'dev' into feat/hmr-indicator
